pkg/machine/e2e/cp_test.go

```diff
@@ -0,0 +1,172 @@
+package e2e_test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("podman machine cp", func() {
+	It("cp all tests", func() {
+		// HOST FILE SYSTEM
+		// ~/<ginkgo_tmp>
+		//   * foo.txt
+		//   - foo-dir
+		//     * bar.txt
+		//
+		//   * guest-foo.txt
+		//   - guest-foo-dir
+		//     * bar.txt
+
+		// GUEST FILE SYSTEM
+		// ~/
+		//   * foo.txt
+		//   - foo-dir
+		//     * bar.txt
+
+		var (
+			file     = "foo.txt"
+			filePath = filepath.Join(GinkgoT().TempDir(), file)
+
+			directory     = "foo-dir"
+			directoryPath = filepath.Join(GinkgoT().TempDir(), directory)
+
+			fileInDirectory     = "bar.txt"
```

Comment:
For some reason tests are failing during cleanup for me on Windows because this file cannot be removed. The error message is "The process cannot access the file because it is being used by another process.". This can be specific to my machine though.

Reply:
I'm relying on the CI for my windows machine, but I'll keep an eye on it.